File: src/fbdb.java
<<<<<<< HEAD
import com.google.auth.oauth2.GoogleCredentials;
import com.google.firebase.FirebaseApp;
import com.google.firebase.FirebaseOptions;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;

import java.io.FileInputStream;
import java.util.HashMap;

/* firebase_dict
type: reference to database
storage: O(1)
Note: all values are immutable
*/

public class fbdb {
	private DatabaseReference ref;

	//Constructor
	public fbdb(String certificate, String url) {
		try {
			FileInputStream serviceAccount = new FileInputStream(certificate);
			FirebaseOptions options = new FirebaseOptions.Builder()
					.setCredentials(GoogleCredentials.fromStream(serviceAccount))
					.setDatabaseUrl(url)
					.build();
			FirebaseApp.initializeApp(options);
			this.ref = FirebaseDatabase.getInstance().getReference();
		} catch (Exception e) {
			System.out.printf("exception");
		}
	}

	public fbdb(DatabaseReference ref) {
		this.ref = ref;
	}
}
public class firebase_dict {
	private DatabaseReference ref;

	//Constructor
	public firebase_dict (String certificate, String url){
		FileInputStream serviceAccount = new FileInputStream(certificate);
		FirebaseOptions options = new FirebaseOptions.Builder()
			.setCredentials(GoogleCredentials.fromStream(serviceAccount))
			.setDatabaseUrl(url)
			.build();
		FirebaseApp.initializeApp(options);
		ref = FirebaseDatabase.getInstance().getReference();
	}
	
	//get value
	public Object get (String key){
		return ref.child(key).getKey();
		return ref.child(key).getValue();
	}
	
	//set value
	public void set (String key, String value){
		ref.child(key).setValue(value);
	}
	
	//delete value 
	public void del (String key){
		ref.child(key).removeValue();
	}

	public fbdb child(String key){
		return new fbdb(ref.child(key));
	}

	public fbdb father() {return new fbdb(ref.getParent());}
	
	//get whole dataset
	public HashMap call (){
		return ref.getValue();
	}
}

File: src/main/java/servlet/fbdb.py
import firebase_admin, firebase_admin.db

""" firebase_dict
type: reference to database
storage: O(1)
Note: all values are immutable
"""
class firebase_dict (dict):
	#Constructor
	def __init__ (self, certificate, url):
		cred = firebase_admin.credentials.Certificate(certificate)
		firebase_admin.initialize_app(cred, {'databaseURL': url})
		self.ref = firebase_admin.db.reference()
		
	#get value
	def __getitem__(self, key):
		return self.ref.child(str(key)).get()

	#set value
	def __setitem__(self, key, value):
		self.ref.child(str(key)).set(value)

	#delete value
	def __delitem__(self, key):
		self.ref.child(str(key)).delete()

	#get whole dataset
	def __call__ (self):
		return self.ref.get()
		
	#Display
	def __repr__ (self):
		return self().__repr__()

if __name__ == "__main__":
	certificate = 'temp-c0b09-firebase-adminsdk-fqxrx-ab2b9d1f13.json'
	url = 'https://temp-c0b09.firebaseio.com/'
	fbd = firebase_dict(certificate, url)
	fbd[100] = [1,2,3]
	print(fbd[100])
	del fbd[100]
	print(fbd)